op-program/client/l2/engineapi/precompiles.go

```diff
@@ -66,6 +73,75 @@ func CreatePrecompileOverrides(precompileOracle PrecompileOracle) vm.PrecompileO
 			return &precompile
 		case kzgPointEvaluationPrecompileAddress:
 			return &kzgPointEvaluationOracle{Orig: orig, Oracle: precompileOracle}
+		case blsG1AddPrecompileAddress:
+			// no size limit - fixed input
+			return &blsOperationOracle{
+				Orig:              orig,
+				Oracle:            precompileOracle,
+				checkInputSize:    checkInputExactSize(256),
+				checkOutput:       checkOutputExactSize(128),
+				precompileAddress: blsG1AddPrecompileAddress,
+			}
+		case blsG1MSMPrecompileAddress:
+			return &blsOperationOracleWithSizeLimit{
+				sizeLimit: 513760,
+				blsOperationOracle: blsOperationOracle{
+					Orig:              orig,
+					Oracle:            precompileOracle,
+					checkInputSize:    checkInputSizeMultipleOf(160),
+					checkOutput:       checkOutputExactSize(128),
+					precompileAddress: blsG1MSMPrecompileAddress,
+				},
+			}
+		case blsG2AddPrecompileAddress:
+			// no size limit - fixed input
+			return &blsOperationOracle{
+				Orig:              orig,
+				Oracle:            precompileOracle,
+				checkInputSize:    checkInputExactSize(512),
+				checkOutput:       checkOutputExactSize(256),
+				precompileAddress: blsG2AddPrecompileAddress,
+			}
+		case blsG2MSMPrecompileAddress:
+			return &blsOperationOracleWithSizeLimit{
+				sizeLimit: 488448,
```

Comment:
TODO to replace this with protocol constants in op-geth.

Reply:
8061a3e